src/sentry/api/event_search.py

```diff
@@ -952,7 +952,8 @@ def convert_search_filter_to_snuba_query(search_filter, key=None, params=None):
                 return [["isNull", [name]], search_filter.operator, 1]
 
         is_null_condition = None
-        if search_filter.operator == "!=" and not search_filter.key.is_tag:
+        # TODO(wmak): Skip this for all non-nullable keys not just event.type
```

Comment:
This is the optimization that was mentioned so we can skip reading an extra `null` mask in clickhouse right?

Reply:
As a follow-up, we can potentially apply this to all `has` queries on non-nullable fields.

Reply:
:+1: https://getsentry.atlassian.net/browse/VIS-751 Going to be the first thing I tackle once we get the trace view to a decent place